### DISCUSSION TITLE
Is it possible to set the session cookies manually? (to bypass login)

### DISCUSSION THREAD

**Discussion by alessandrojcm** [nextauthjs/next-auth#3613] [Help]
Hi there folks, I've got this kinda very specific use-case I'm trying to solve out; but I can't seem to find a way around it. 

A little background first so you can see where I come from: at my work we switched to FusionAuth as our IdP not so long ago across all of our stack (client-facing apps both web and mobile as well as our administration portal). Now that we've made that switch, I'm looking for a way to implement impersonation for our support team. FusionAuth does not support impersonation per-se, but one of their developers suggests a [way](https://github.com/FusionAuth/fusionauth-issues/issues/830#issuecomment-685168794) of implementing it. The issue I've got specifically is that while the app I want to implement impersonation for is using `next-auth`, the administration portal is built with PHP Laravel, and also they're hosted on different domains.

So the idea I had of how to implement his feature is something like this:
1. Add an "impersonate user" button in the user's details page in the administration portal (which is using oauth2 for login with 2fa)
2. That button will generate a very short-lived [JWT](https://fusionauth.io/docs/v1/tech/apis/jwt/) with the id for the user to be impersonated and some other payload for verification purposes
3. It would redirect to a Next API route protected by several methods (`Authorization` header, only allowing requests from the admin domain, etc)
4. It would create the authorization token as per the issue I linked earlier and call FusionAuth's login method to get an access and refresh token
5. It would invalidate the aforementioned JWT token so it can only be used once
6. Somehow I would need to set the session cookie here for `next-auth`, so it considers that the user is logged in.

Now, my problem lies in the last point. AFAIK there's no way in `next-auth` to log the user in from the server without doing the whole log in flow, and the API for setting the session cookies is not public (AFAIK). Is there I way I could accomplish this? (besides copy-pasting the code from the library); is this even a good idea or do you think there's a better way of accomplishing this? Don't be afraid to call me out if you think I'm crazy :sweat_smile: 

Thanks in advance

**Comment by alessandrojcm** [comment]
Ok so I've come into this https://github.com/nextauthjs/next-auth/discussions/3133; maybe I can work with that.

**Reply by maxeth** [reply]
I'm trying to accomplish something almost identical to your use case and don't quite understand how this thread relates to the issue.

Did you end up finding a way to manually replicate the auth-flow with an arbitrary JWT payload? 🙂

**Reply by alessandrojcm** [reply]
> I'm trying to accomplish something almost identical to your use case and don't quite understand how this thread relates to the issue.
> 
> Did you end up finding a way to manually replicate the auth-flow with an arbitrary JWT payload? 🙂

Hi there, sorry but I don't work at the company I was trying to implement this feature for anymore thus I didn't keep working on it.
